pkg/gatewayserver/gatewayserver.go

```diff
@@ -246,7 +246,11 @@ func New(c *component.Component, conf *Config, opts ...Option) (gs *GatewayServe
 	if conf.BasicStation.FallbackFrequencyPlanID != "" {
 		bsCtx = frequencyplans.WithFallbackID(bsCtx, conf.BasicStation.FallbackFrequencyPlanID)
 	}
-	bsWebServer := basicstationlns.New(bsCtx, gs, conf.BasicStation.UseTrafficTLSAddress, conf.BasicStation.WSPingInterval)
+	bsWebServer := basicstationlns.New(bsCtx, gs, basicstationlns.Options{
```

Comment:
At this point you can move the `BasicStationConfig` from `pkg/gatewayserver/config.go` to the `basicstationlns` package, and accept it in `New`.

Reply:
Yeah good suggestion. Added that now.